src/charm.py

```diff
@@ -671,6 +662,30 @@ def _on_rotate_key_action(self, event: ActionEvent) -> None:
         event.log("Successfully rotated the JWK")
         event.set_results({"new-key-id": jwk_id})
 
+    @leader_unit
+    def _clean_up_oauth_relation_clients(self):
+        for k in list(self.peer_data.keys()):
+            if not k.startswith("oauth_"):
+                continue
+
+            rel_id = k[len("oauth_") :]
+            if self.model.get_relation(OAUTH_INTEGRATION_NAME, relation_id=int(rel_id)).active:
```

Comment:
Question: Would it be enough to just run this logic on `client_deleted`?
According to https://ops.readthedocs.io/en/latest/reference/ops.html#ops.Relation.active this is only inactive on relation broken events

Reply:
Good catch, I think that this means that this shouldn't be run on `client_deleted`, as the relation will be departing then.